tests/expected/function-contract/modifies/expr_pass.rs

```diff
@@ -7,7 +7,7 @@
 
 #[kani::requires(**ptr < 100)]
 #[kani::modifies(ptr.as_ref())]
-#[kani::ensures(**ptr < 101)]
+#[kani::ensures(|result| **ptr < 101)]
```

Comment:
This can be implemented by capturing the failure of parsing the ensures statement as a closure, then assuming it is a boolean expression that doesn't include the result. There is no such thing as a function with no return as it just returns unit, but this change can be done for any ensures statement that doesn't refer to the result within the body of the closure.
However, it is probably best left as is as this would standardize the ensures statement